### ISSUE TITLE
Connecting to Pusher throws ClassCastException

### ISSUE THREAD

**Issue by Maveriks** [pusher/pusher-websocket-java#155]
### The Issue

Cannot connect to Pusher because of ClassCastException

### Details

`FATAL EXCEPTION: pusher-java-client eventQueue
                                                 Process: com.orthinc.orthinc, PID: 8643
                                                 com.google.gson.JsonParseException: The JsonDeserializer MapTypeAdapter failed to deserialize json object {"event":"pusher:connection_established","data":"{\"socket_id\":\"123655.3303525\"}"} given the type interface java.util.Map`

From this issue #41 it looks as though this isn't the json that should be returned. However I am not connecting to a custom host.

**EDIT**

My mistake. 

After re-reading the previous issue it looks as though this is the expected JSON so the issue must be with the client. I will post the full stack trace as a comment

---- 

CC @pusher/mobile

**Comment by zmarkan** [comment]
Hi,  that does indeed seem odd. 
The response seems fine to me. Can you give some more details, namely: 

- Does it happen every time, on every device?
- Are you using the latest version of the library - `1.5.0` 
- Are you forcing a specific version of the Gson dependency by any chance? 

Thanks

**Comment by Maveriks** [comment]
```
FATAL EXCEPTION: pusher-java-client eventQueue
                                                                     Process: com.orthinc.orthinc, PID: 15980
                                                                     com.google.gson.JsonParseException: The JsonDeserializer MapTypeAdapter failed to deserialize json object {"event":"pusher:connection_established","data":"{\"socket_id\":\"123643.3337687\"}"} given the type interface java.util.Map
                                                                         at com.google.gson.JsonDeserializerExceptionWrapper.deserialize(JsonDeserializerExceptionWrapper.java:64)
                                                                         at com.google.gson.JsonDeserializationVisitor.invokeCustomDeserializer(JsonDeserializationVisitor.java:92)
                                                                         at com.google.gson.JsonDeserializationVisitor.visitUsingCustomHandler(JsonDeserializationVisitor.java:80)
                                                                         at com.google.gson.ObjectNavigator.accept(ObjectNavigator.java:101)
                                                                         at com.google.gson.JsonDeserializationContextDefault.fromJsonObject(JsonDeserializationContextDefault.java:76)
                                                                         at com.google.gson.JsonDeserializationContextDefault.deserialize(JsonDeserializationContextDefault.java:54)
                                                                         at com.google.gson.Gson.fromJson(Gson.java:551)
                                                                         at com.google.gson.Gson.fromJson(Gson.java:498)
                                                                         at com.google.gson.Gson.fromJson(Gson.java:467)
                                                                         at com.google.gson.Gson.fromJson(Gson.java:417)
                                                                         at com.google.gson.Gson.fromJson(Gson.java:389)
                                                                         at com.pusher.client.connection.websocket.WebSocketConnection$6.run(WebSocketConnection.java:256)
                                                                         at com.pusher.client.util.Factory$1.run(Factory.java:103)
                                                                         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                         at java.lang.Thread.run(Thread.java:761)
                                                                      Caused by: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
                                                                         at com.google.gson.internal.$Gson$Types.getMapKeyAndValueTypes($Gson$Types.java:318)
                                                                         at com.google.gson.MapTypeAdapter.deserialize(MapTypeAdapter.java:65)
                                                                         at com.google.gson.MapTypeAdapter.deserialize(MapTypeAdapter.java:33)
                                                                         at com.google.gson.JsonDeserializerExceptionWrapper.deserialize(JsonDeserializerExceptionWrapper.java:51)
```

**Comment by Maveriks** [comment]
@zmarkan 

Hi, thanks for the quick reply! In answer to your questions

- This happens on every android device I have tested it on, including a physical device. However we have the iOS Swift library working.
- We are using version 1.5.0
- I don't think we are forcing a specific version

**Comment by zmarkan** [comment]
Hm, interesting. 

Can you paste a snippet of the offending code please.
If you prefer not to share your code here you can also open a support ticket, and mention this Github issue - https://support.pusher.com/hc/en-us/requests/new

Then I can look into it for you, and try to reproduce it.

Thanks

**Comment by Maveriks** [comment]
I managed to fix my issue here.

Looks like you were almost certainly right with "Are you forcing a specific version of the Gson dependency by any chance?".

I removed all but default dependencies and cleaned + built and then re-added the dependencies one at a time. Now it works! I tried to clean + rebuild before but it didn't solve the issue so I really can't tell what caused this issue.

Many thanks for your help here.
